pkg/nmstate/policy.go

```diff
@@ -264,6 +264,73 @@ func (builder *PolicyBuilder) WithBondInterface(slavePorts []string, bondName, m
 	return builder.withInterface(newInterface)
 }
 
+// WithVlanInterface adds VLAN interface configuration to the NodeNetworkConfigurationPolicy.
+func (builder *PolicyBuilder) WithVlanInterface(baseInterface string, vlanID uint16) *PolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
+
+		return builder
+	}
+
+	glog.V(100).Infof("Creating NodeNetworkConfigurationPolicy %s with VLAN interface configuration: %s %d",
+		builder.Definition.Name, baseInterface, vlanID)
+
+	if baseInterface == "" {
+		glog.V(100).Infof("The baseInterface can not be empty string")
+
+		builder.errorMsg = "The baseInterface is empty sting"
+	}
+
+	if vlanID > 4094 {
+		builder.errorMsg = "invalid vlanID, allowed vlanID values are between 0-4094"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	newInterface := NetworkInterface{
+		Name:  fmt.Sprintf("%s.%d", baseInterface, vlanID),
+		Type:  "vlan",
+		State: "up",
+		Vlan: Vlan{
+			BaseIface: baseInterface,
+			ID:        int(vlanID),
+		},
+	}
+
+	return builder.withInterface(newInterface)
+}
+
+// WithAbsentInterface appends the configuration for an absent interface to the NodeNetworkConfigurationPolicy.
+func (builder *PolicyBuilder) WithAbsentInterface(interfaceName string) *PolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
```

Comment:
No need to do this, the validate method does it automatically

Reply:
removed